pyccl/halos/profiles.py

```diff
@@ -1551,3 +1554,262 @@ def _Ns(self, M, a):
+        '''
+        Computes the angular part of the satellite intrinsic shear field,
+        Eq. (C8) in Fortuna et al. 2021.
+        '''
+        from scipy.special import binom
+        gj = np.array([0, 0, np.pi / 2, 0, np.pi / 2, 0, 15 * np.pi / 32,
+                       0, 7 * np.pi / 16, 0, 105 * np.pi / 256, 0,
+                       99 * np.pi / 256])
+        l_sum = 0.
+        if b == 0:
+            var1_add = 1
+        else:
+            var1_add = b
+        for m in range(0, l + 1):
+            m_sum = 0.
+            for j in range(0, m + 1):
+                m_sum += (binom(m, j) * gj[j] *
+                          self._I_integral(j + var1_add, m - j) *
+                          np.sin(thk) ** j * np.cos(thk) ** (m - j))
+            l_sum += binom(l, m) * binom((l + m - 1) / 2, l) * m_sum
+        if phik is not None:
+            l_sum *= np.exp(1j * 2 * phik)
+        return 2 ** l * l_sum
+
+    def gamma_I(self, r, r_vir):
+        '''
+        Returns the intrinsic satellite shear,
+        .. math::
+            \\gamma^I(r)=a_{1\\mathrm{h}}
+            \\left(\\frac{r}{r_\\mathrm{vir}}\\right)^b.
+        If :math:`b` is 0, then only the value of the amplitude
+        :math:`a_\\mathrm{1h}` is returned. In addition, according to
+        Fortuna et al. 2021, we use a constant value of 0.06 Mpc for
+        :math:`r<0.06` Mpc and set a maximum of 0.3 for :math:`\\gamma^I(r)`.
+        '''
+        if self.b == 0:
+            return self.a1h
+        else:
+            r_use = np.copy(np.atleast_1d(r))
+            r_use[r_use < 0.06] = 0.06
+            if np.ndim(r_vir == 1):
+                r_vir = r_vir.reshape(len(r_vir), 1)
+            # Do not output value higher than 0.3
+            gamma_out = self.a1h * (r_use/r_vir) ** self.b
+            gamma_out[gamma_out > 0.3] = 0.3
+            return gamma_out
+
+    def _rvir(self, cosmo, M, a, mdef):
+        """
+        This translates a halo mass into a comoving radius,
+        returns the comoving virial radius (in Mpc).
+        """
+        return mdef.get_radius(cosmo, M, a) / a
+
+    def _real(self, cosmo, r, M, a, mass_def):
+        '''
+        Returns the real part of the satellite intrinsic shear field,
+        .. math::
+            \\gamma^I(r) u(r|M),
+        with :math:`u` being the halo density profile divided by its mass.
+        For now, it assumes a NFW profile.
+        '''
+        M_use = np.atleast_1d(M)
+        r_use = np.atleast_1d(r)
+
+        rvir = self._rvir(cosmo, M_use, a, mass_def)
+        # Density profile from HOD class - truncated NFW
+        u = self._usat_real(cosmo, r_use, M_use, a, mass_def)
+        prof = self.gamma_I(r_use, rvir) * u
+
+        if np.ndim(r) == 0:
+            prof = np.squeeze(prof, axis=-1)
+        if np.ndim(M) == 0:
+            prof = np.squeeze(prof, axis=0)
+
+        return prof
+
+    def _usat_fourier(self, cosmo, k, M, a, mass_def):
```

Comment:
OK, I'm trying to understand the logic for this one. So you are overloading the `_usat_fourier` method in the HOD profile so that it now returns the satellite shear profile, right?

But then, this will go into [this line](https://github.com/LSSTDESC/CCL/blob/d5408a777ab5fc4b2ae6fada2114a168b9bacee4/pyccl/halos/profiles.py#L1507), which also contains a contribution from centrals to the total number density. What's happening with that contribution?

Reply:
That is correct. The contribution from centrals is going to be neglected because of [this line](https://github.com/LSSTDESC/CCL/blob/1ce402c63ca7b01c739b32c55c1fdfce7450a922/pyccl/halos/profiles.py#L1634), where the `fc_0=0` and `ns_independent=True`.

The idea behind this is that, for the alignments between centrals we can use the NLA model. There might be a better way to do this, however, so I am happy to discuss this.